### DISCUSSION TITLE
Naming

### DISCUSSION THREAD

**Discussion by mishaglik** [mishaglik/Elpidifor-s-legacy#12] [Ideas]
# Naming is very confusing
We have very bad naming. It seriously confuses users of standart. So we should create better one. Here we will collect all naming issues.

**Comment by mishaglik** [comment]
#10 putPixel -> setPixel

**Reply by XDLoLiK** [reply]
mc Ψna против

**Comment by mishaglik** [comment]
Canvas is most confusing. Will it be kept for compatibility or rename?

**Reply by Lord-KA** [reply]
Maybe `customWidget`? (Or `selfhostedWidget`, but it seems too long.)

**Reply by mishaglik** [reply]
ToolWidget?

**Reply by deGekata** [reply]
F*** Canvas, all my homies use ToolFrame

**Comment by Lord-KA** [comment]
`scrollBar` is a thing that extends a window by providing a `slider` (a long bar with a marker in it to select value in an interval). In 'Canvas' we meant adding a `slider`, not a `scrollBar`.

![slider](https://user-images.githubusercontent.com/43700614/204131395-75d298ab-3b28-4b0b-9bb0-d7483e37a62b.png)
![scrollBar](https://user-images.githubusercontent.com/43700614/204131423-80ebf17b-bfb2-4956-b868-728da5cc7dc0.png)
